### PR TITLE
sanitized url in url Validation

### PR DESCRIPTION
**Pull request by Sooraj-s-98**
## What does this PR do?

In Germany, we are allowed to use Umlauts as ä, ö or ü in a URL. But the validation filters it out.

> currently 
> Note that the function will only find ASCII URLs to be valid; internationalized domain names (containing non-ASCII characters) will fail.

## Test Plan

`https://www.amazon.de/AMAZONAS-Hängemattengestell-RockStone-größenverstellbar-270-330/dp/B005HGW6A4`
valid URL .

## Related PRs and Issues

[Bug Report: Document creation failed when using url attribute](https://github.com/appwrite/appwrite/issues/3670)

### Have you read the [Contributing Guidelines on issues](https://github.com/appwrite/appwrite/blob/master/CONTRIBUTING.md)?

yes

### CODE REVIEW

**Comment by TorstenDittmann** [comment]
Would you mind adding a test for this here => https://github.com/appwrite/appwrite/blob/master/tests/unit/Network/Validators/URLTest.php 🙏🏻

**Comment by Sooraj-s-98** [comment]
@TorstenDittmann  sure

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Could you please update your branch with our latest master? It seems your branch is missing some files for our CI.

**Comment by Sooraj-s-98** [comment]
@stnguyen90  can you check

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
Looks like the test case failed.

**Comment by Sooraj-s-98** [comment]
@stnguyen90  can you check?

File: src/Appwrite/Network/Validator/URL.php [hunk 1]
```diff
@@ -49,11 +49,17 @@ public function getDescription(): string
      */
     public function isValid($value): bool
     {
-        if (\filter_var($value, FILTER_VALIDATE_URL) === false) {
+        $sanitized_url = '';
+
+        foreach (str_split($value) as $character) {
+            $sanitized_url .= (ord($character) > 127) ? rawurlencode($character) : $character;
```
Comment (Meldiron) [inline src/Appwrite/Network/Validator/URL.php:55]: I am not experienced with this, would it be please possible to include link to a resource that explains why we are looking for `ord() > 127`?

Comment (Sooraj-s-98) [reply src/Appwrite/Network/Validator/URL.php:55]: refer [ doc](https://www.php.net/manual/en/function.ord.php)
@Meldiron

File: src/Appwrite/Network/Validator/URL.php [hunk 2]
```diff
@@ -49,11 +49,17 @@ public function getDescription(): string
      */
     public function isValid($value): bool
     {
-        if (\filter_var($value, FILTER_VALIDATE_URL) === false) {
+        $sanitized_url = '';
```
Comment (stnguyen90) [inline src/Appwrite/Network/Validator/URL.php:52]: Would you please update this to be camel cased to match our convention?

```suggestion
        $sanitizedURL = '';
```

**Comment by stnguyen90** [comment]
Some additional tests:

```
input: https://example.com
sanitized: https://example.com
isValid: 1
-----------------------------------------
input: https://example.com/foo%2\u00c2\u00ä9zbär
sanitized: https://example.com/foo%2\u00c2\u00%C3%A49zb%C3%A4r
isValid: 1
-----------------------------------------
input: https://www.amazon.de/AMAZONAS-Hängemattengestell-RockStone-größenverstellbar-270-330/dp/B005HGW6A4
sanitized: https://www.amazon.de/AMAZONAS-H%C3%A4ngemattengestell-RockStone-gr%C3%B6%C3%9Fenverstellbar-270-330/dp/B005HGW6A4
isValid: 1
-----------------------------------------
```

I also tried browsing to `https://www.amazon.de/AMAZONAS-Hängemattengestell-RockStone-größenverstellbar-270-330/dp/B005HGW6A4` and got the same value as this implementation (`https://www.amazon.de/AMAZONAS-H%C3%A4ngemattengestell-RockStone-gr%C3%B6%C3%9Fenverstellbar-270-330/dp/B005HGW6A4`)

image

**Review by stnguyen90 — CHANGES_REQUESTED** [review]
The implementation looks good to me! Just 1 more style change. Apologies for missing it before! 🙏

File: src/Appwrite/Network/Validator/URL.php (hunk 2 above)
Comment (Sooraj-s-98) [reply src/Appwrite/Network/Validator/URL.php]: sure

**Review by stnguyen90 — APPROVED** [review]
Great job with the PR! 🚀 We've approved the PR and will merge it soon.

**Pull request merged**